### DISCUSSION TITLE
Code cell format, CoCalc, Multiple Kernels etc.

### DISCUSSION THREAD

**Discussion by mforbes** [executablebooks/meta#1137] [Ideas]
The online [CoCalc][] platform provides one of the only truly collaborative (realtime) computing platforms, and has been developing a [Markdown editor][] that allows code cells to be executed.  I have been encouraging them to [add support for some of the myst extensions][], which would turn this into an awesome interactive place to collaboratively edit Jupyter Books.

I would like some advice from the Executable Books team to guide them so that their [markdown editor][] evolves in a way compatible with the MyST NB format. Some relevant issues and discussions:

* The most relevant/timely discussion right now is about [CoCalc][]'s execution model, which supports executing different kernels in the same document. I know there are some issues with this being supported in Jupyter notebooks, but mainly want to make sure that whatever format they choose to specify the kernel is compatible with MyST NB. A couple of issues (related to `thebe`) requesting similar support might be relevant:
  * https://github.com/executablebooks/thebe/issues/79
  * https://github.com/executablebooks/thebe/issues/343

Would anyone from the Executable Books team be willing to chat a bit about this and/or help work with the CoCalc team to ensure that consistent design choices are made? 

## Relevant Discussion

Open [CoCalc][] issues and discussions:
* https://github.com/sagemathinc/cocalc/issues/5948
* https://github.com/sagemathinc/cocalc/issues/7193
* https://github.com/sagemathinc/cocalc/issues/4715

Some related Executable Books issues:
* https://github.com/executablebooks/thebe/issues/79
* https://github.com/executablebooks/thebe/issues/343

[add support for some of the myst extensions]: <https://github.com/sagemathinc/cocalc/issues/5948>
[CoCalc]: <https://doc.cocalc.com/markdown.html>
[Markdown editor]: <https://doc.cocalc.com/markdown.html>

**Comment by williamstein** [comment]
Just putting a comment here to say that I'm interested in this discussion, and willing to make changes to our current implementation and design choices.

**Reply by rowanc1** [reply]
Hi @williamstein and @mforbes thanks so much for opening up the discussion here - I would love to be involved in a discussion on how the EBP and MyST communities can be better integrated into the projects you are working on. I think that we have some really awesome building blocks here that could hopefully be helpful to you!

I think it would be helpful to have @stevejpurves and @agoose77 in a preliminary discussion -- I reached out by email to you to find a time. Looking forward to it. :)
